### ISSUE TITLE
Error on gopls start

### ISSUE THREAD

**Issue by yuriizinets** [lite-xl/lite-xl-lsp#43]
LSP config:
```lua
local lspconfig = require "plugins.lsp.config"

...

lspconfig.gopls.setup {
  command = {
    "gopls"
  },
  settings = {
    Lua = {
      diagnostics = {
        enable = false
      }
    }
  },
  verbose = true
}
```

<img width="1552" alt="Screenshot 2022-12-28 at 11 31 06" src="https://user-images.githubusercontent.com/17050536/209798583-11e460c6-874c-48a8-b120-4f9686dcd09d.png">

**Comment by yuriizinets** [comment]
Updated config with absolute gopls path, now getting this:

<img width="1552" alt="image" src="https://user-images.githubusercontent.com/17050536/209923075-ac02989a-9e58-4a7d-b46c-8fc0992acd2d.png">

**Comment by jgmdev** [comment]
I think the issue is you are adding settings that are specifically for the lua language server:
```lua
lspconfig.gopls.setup {
  command = {
    "gopls"
  },
  -- settings = { -- these settings are for the lua language server not gopls
  --  Lua = {
  --    diagnostics = {
  --      enable = false
  --    }
  --  }
  -- },
  verbose = true
}
```
A simple setup call to `lspconfig.gopls.setup()` should be enough to enable the go language server.

**Comment by yuriizinets** [comment]
With a generic `lspconfig.gopls.setup()` I have this issue, that's why I've ended up with explicit args.

<img width="692" alt="image" src="https://user-images.githubusercontent.com/17050536/210225271-02b15a29-769c-4e5d-8271-a54cb3937d26.png">

**Comment by jgmdev** [comment]
Could be that your `gopls` is not the direct binary but a bash/shell script that points to it which makes the lsp properly detect the real process id of running lsp server. Try from terminal with `which gopls` to see the path of invoked gopls binary and then confirm it is the real gopls binary and not a shell script invoking it.

**Comment by yuriizinets** [comment]
Right, it was a soft linked into `/usr/local/bin/gopls` as far as binary located in `~/go/bin` and can't be located by editor.

Also, tried this approach:
```
lspconfig.gopls.setup {
  command = {
    "/Users/yuriizinets/go/bin/gopls"
  }
}
```

Getting this error

<img width="1128" alt="image" src="https://user-images.githubusercontent.com/17050536/210520279-7b0df3fa-6076-46d4-b862-d1fcda0f6bc4.png">

**Comment by calvinweb** [comment]
pylsp and pyright meets the same error

**Comment by Guldoman** [comment]
Is this still an issue?
